### ISSUE TITLE
feat: group on multiple columns

### ISSUE THREAD

**Issue by ryanzomorrodi** [ryanzomorrodi/centr#15]
I could imagine having a dataset like:
```r
#>   STATEFP COUNTYFP TRACTFP
#> 1 01      133      965700
#> 2 02      133      965700
```
I would concatenate these strings together, but I could imagine another scenario like 

```r
#>   STATEFP COUNTYFP TRACTFP DATE
#> 1 01      133      965700  2024/01/01
#> 2 02      133      965700  2024/01/01
```
If I wanted to get groups for each tract on each date, I'd probably want to do a 
```r
library(dplyr)
library(purrr)

df |>
    group_split(DATE) |>
    mutate(fips = paste0(STATEFP, COUNTYFP, TRACTFP)) |>
    group_map(~ mean_center(.x, "fips", "some_weight"))
```
And then I'd have to add the dates back in and bind the dfs together.

Grouping on multiple columns, would make this quite a bit easier.
```r
mean_center(df, group = c("STATEFP", "COUNTYFP", "TRACTFP", "DATE"), "some_weight")
```

**Comment by ryanzomorrodi** [comment]
From a cursory testing, `group_by` is considerably faster than split. Not sure if its worth taking on a dplyr dependency

**Comment by ryanzomorrodi** [comment]
See [18e9509](https://github.com/ryanzomorrodi/centr/commit/18e95090d62d64b0cee7922eeb95e59dd83269c9)

**Comment by ryanzomorrodi** [comment]
Fixed by 68a3e1c
